src/samples/audio/smart_amp_test.c

```diff
@@ -555,14 +830,22 @@ static const struct comp_driver comp_smart_amp = {
 	.uid = SOF_RT_UUID(smart_amp_comp_uuid),
 	.tctx = &smart_amp_comp_tr,
 	.ops = {
-		.create = smart_amp_new,
-		.free = smart_amp_free,
-		.params = smart_amp_params,
-		.prepare = smart_amp_prepare,
-		.cmd = smart_amp_cmd,
-		.trigger = smart_amp_trigger,
-		.copy = smart_amp_copy,
-		.reset = smart_amp_reset,
+		.create			= smart_amp_new,
+		.free			= smart_amp_free,
+		.params			= smart_amp_params,
+		.prepare		= smart_amp_prepare,
+#if CONFIG_IPC_MAJOR_4
+		.set_large_config	= smart_amp_set_large_config,
+		.get_large_config	= smart_amp_get_large_config,
+		.get_attribute		= smart_amp_get_attribute,
+		.bind			= smart_amp_bind,
+		.unbind			= smart_amp_unbind,
```

Comment:
@RanderWang I keep repeating this every time a module is converted to IPC4. bind/unbind are not compatible with the module adapter interface. If we keep adding this si every new component with IPC4, it will take us forever to deprecate comp_driver. Can we please find a way to not do this and make our own lives difficult later on?

Reply:
@ranj063 this interface is efficient a way to deal with event.  Without it , what I can figure out is that in each smart copy or smart trigger we need to check whether there is a feedback buffer connected or disconnected. I don't like this solution. If we can add a event processor for each module, we can also support this.

Reply:
@RanderWang don't we have notifications for this?

Reply:
@ranj063  sure. If it is not fit for module_adapter, we need to remove comp_bind interface and use notification to replace it (a new notification like MODULE_BIND/UNBIND). I found that comp_bind is used by mixin_out. I don't want to mix the rework with this smart_amp_test PR. Can I submit the rework in another PR ?